### PR TITLE
Distributed Covariance 

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**

### CODE REVIEW

File: cpp/oneapi/dal/backend/primitives/stat/cov_dpc.cpp [hunk 1]
```diff
@@ -363,6 +362,185 @@ sycl::event correlation_with_covariance(sycl::queue& q,
     return finalize_event;
 }
 
+template <typename Float>
+inline sycl::event finalize_covariance_distributed(sycl::queue& q,
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/backend/primitives/stat/cov_dpc.cpp:366]: The same as finalize_covariance(...) ?

File: cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_impl_dpc.cpp [hunk 2]
```diff
@@ -0,0 +1,858 @@
...
+
+template <typename Float>
+auto compute_correlation(sycl::queue& q,
+                         std::int64_t row_count,
+                         const pr::ndview<Float, 2>& xtx,
+                         const pr::ndarray<Float, 1>& sums,
+                         const dal::backend::event_vector& deps = {}) {
+    ONEDAL_PROFILER_TASK(compute_correlation, q);
+    //ONEDAL_ASSERT(data.has_data());
+    //ONEDAL_ASSERT(data.get_dimension(1) == sums.get_dimension(0));
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_impl_dpc.cpp:131]: Please uncomment

File: cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_impl_dpc.cpp [hunk 3]
```diff
@@ -0,0 +1,858 @@
...
+    const std::int64_t column_count = xtx.get_dimension(1);
+    auto corr =
+        pr::ndarray<Float, 2>::empty(q, { column_count, column_count }, sycl::usm::alloc::device);
+    copy(q, corr, xtx, { deps }).wait_and_throw();
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/covariance/backend/gpu/compute_kernel_dense_impl_dpc.cpp:135]: Why do u wait here but not passing it throw other kernels?

**Review by KulikovNikita — APPROVED** [review]

**Review by napetrov — APPROVED** [review]

**Review by amgrigoriev — APPROVED** [review]

**Pull request merged**